src/rdkafka_transport.c

```diff
@@ -498,6 +500,14 @@ static int rd_kafka_transport_ssl_connect (rd_kafka_broker_t *rkb,
 	if (!SSL_set_fd(rktrans->rktrans_ssl, rktrans->rktrans_s))
 		goto fail;
 
+	mtx_lock(&rkb->rkb_lock);
+	rd_snprintf(name, sizeof(name), "%s", rkb->rkb_nodename);
+	mtx_unlock(&rkb->rkb_lock);
+	if ((t = strrchr(name, ':')))
+		*t = '\0';
+	if (!SSL_set_tlsext_host_name(rktrans->rktrans_ssl, name))
```

Comment:
So, I have looked a bit what was the most common way to detect this feature.

There is:

 - `#if (OPENSSL_VERSION_NUMBER >= 0x0090806fL) && !defined(OPENSSL_NO_TLSEXT)`
 - `#if !defined(OPENSSL_NO_TLSEXT) && defined(SSL_set_tlsext_host_name)`
 - `#ifdef SSL_CTRL_SET_TLSEXT_HOSTNAME`

The first one seems more future proof for me, as if the implementation changes, we'll still get the feature implemented. The other cases rely on the current implementation. Alternate SSL implementation, like LibreSSL, also define `OPENSSL_VERSION_NUMBER`, sometimes with a bogus value large enough for our use.

So, I'll amend the commit with this change.

Another question: many implementations only set hostname for non-numeric value. Do you see a quick way to test we don't have just an IP address? If not, I don't think it's worth writing additional code for that.